### DISCUSSION TITLE
Continue if not working on Define Actions

### DISCUSSION THREAD

**Discussion by juangbm** [Budibase/budibase#9896] [Help] [answered]
Hi,

I am using an action "Continue if" when defining an action from a Save button, and it is not working, I have to make a REST call and based on the result code returned I need to continue saving or stop.
I am using this setting:
![image](https://user-images.githubusercontent.com/22911293/223183969-08a5607b-ff7c-487f-8242-90e8375ec53a.png)
and it never continues, even when I am getting 0 in the code field.
I am Showing a notification **before** the "Continue if" block and I can see that 0 value is being returned, but no way to execute what it is after the "Continue if" Action
![image](https://user-images.githubusercontent.com/22911293/223185228-83ac4f32-02c2-4232-b7d8-8a8bff0fe57a.png)

![image](https://user-images.githubusercontent.com/22911293/223184841-5d68d12f-8ee1-45f1-aca5-8766ae792d45.png)

What can be going on?

Thanks
Juan

**Comment by andz-bb** [comment, marked as answer]
Hey @juangbm, this is a strange one! I've tried it out on my end and I have been unable to recreate what you're seeing. Could you send me an example of the full query response found when testing it in your data tab and I'll see if I can recreate with that. 

Alternatively as a workaround you could try and do the check completely within the javascript binding, and then see if it evaluates to true or false. Console logs are helpful here to see the type of the value on the response if that's what's causing the problem. JS can be weird sometimes with how it treats values like 0 so this might clear it up a bit.

<img width="2137" alt="image" src="https://user-images.githubusercontent.com/110921612/223204299-31e495fb-1be8-4ffb-9441-919b72729883.png">

**Reply by juangbm** [reply]
Hi @andz-bb 
This is a query response I am using:
{
	"data": [
		{
			"code": 0,
			"message": "OK",
			"data": {
				"name": "RE Juan 1",
				"id": 180,
				"slug": "re-juan-1"
			}
		}
	],	
	"headers": {
		"date": "Mon, 06 Mar 2023 12:21:23 GMT",
		"content-type": "application/json; charset=utf-8",
		"content-length": "81",
		"connection": "close",
		"server": "Kestrel"
	}
}

Thanks!

**Reply by andz-bb** [reply]
hey @juangbm, thanks for sending that through! Would it be possible for you to send an export of your app? From everything I've seen here I would expect the Continue if action to work properly. 

I've attached an app export of the app I've been using to test this out if that's helpful for you to see:
[Mocha REST-export-1678203157597.tar.gz](https://github.com/Budibase/budibase/files/10911129/Mocha.REST-export-1678203157597.tar.gz)

**Reply by melohagan** [reply]
@juangbm 
It might be worth trying JavaScript `return 0` rather than the '0' string in your first screenshot.

It may be incorrectly doing a `0 === "0"` behind the scenes. 

If that quick fix doesn't work then what Andy suggested is a good call!

**Reply by juangbm** [reply]
HI @melohagan,

I just tried your solution, but it doesn't work ... 
I made it work using Javascript @andz-bb proposed solution.

Thanks!

**Reply by NicolasFabre81** [reply]
I'm just having the same problem : my actions are in a condition block.
My JS function does return 'oui' but it does not continue :
<img width="395" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/9c639081-6cf1-427d-95d2-39091661bf3a">

**Reply by NicolasFabre81** [reply]
I think it is related to the condition : I moved my actions from condition to normal onclick actions : it works.
